tests/admin_views/tests.py

```diff
@@ -4688,6 +4718,28 @@ def test_readonly_backwards_ref(self):
         response = self.client.get(reverse('admin:admin_views_topping_add'))
         self.assertEqual(response.status_code, 200)
 
+    def test_readonly_onetoone_backwards_ref(self):
+        """
+        Regression test for OneToOneField backward reference, when related field is read-only.
+        """
+        v1 = Villain.objects.create(name='Adam')
+        pl = Plot.objects.create(name='Test Plot', team_leader=v1, contact=v1)
+        pd = PlotDetails.objects.create(details='Brand New Plot', plot=pl)
+
+        response = self.client.get(reverse('admin:admin_views_plotproxy_change', args=(pl.pk,)))
+        plot_details_field = self.get_admin_readonly_field(response.context['adminform'], 'plotdetails')
```

Comment:
Ok. Passed `admin_form` initially into the method, since it was not intended to extract anything else from response, other than the form.